src/wasm/wasm-ir-builder.cpp

```diff
@@ -175,7 +175,7 @@ Result<Expression*> IRBuilder::pop(size_t size) {
   CHECK_ERR(packageHoistedValue(*hoisted, size));
 
   auto* ret = scope.exprStack.back();
-  if (ret->type.size() == size) {
+  if (ret->type.size() == size || ret->type == Type::unreachable) {
```

Comment:
From your description I thought this was for tuples, but `ret->type.size() == size` can be the case for non-tuples too, can't it? (when `size == 1`)

Reply:
Yes, but since we report the size of `Type::unreachable` as 1, this code path was already taken for singular unreachable popped values.

Reply:
I see, thanks. Maybe worth a comment?